roles/ssl/tasks/provided_keystore_and_truststore.yml

```diff
@@ -1,13 +1,45 @@
 ---
-- name: Copy Host Truststore to Host
+- name: Copy Host Truststore to Host if on control node
   copy:
     src: "{{ssl_truststore_filepath}}"
     dest: "{{truststore_path}}"
+    owner: "{{user}}"
+    group: "{{group}}"
+    mode: 0640
+  when: not ( ssl_provided_keystore_and_truststore_remote_src|bool )
 
-- name: Copy Host Keystore to Host
+- name: Copy Host Keystore to Host if on control node
   copy:
     src: "{{ssl_keystore_filepath}}"
     dest: "{{keystore_path}}"
+    owner: "{{user}}"
+    group: "{{group}}"
+    mode: 0640
+  when: not ( ssl_provided_keystore_and_truststore_remote_src|bool )
+
+- name: Get stats and permissions of keystore
+  ansible.builtin.stat:
+    path: "{{keystore_path}}"
+  register: st_keystore
+
+- name: Get stats and permissions of truststore
+  ansible.builtin.stat:
+    path: "{{truststore_path}}"
+  register: st_truststore
+
+- name: Validate keystore permissions
+  debug:
+    msg:  >-
+      "Ensure that the keystore has valid permissions. The unix file permission of the keystore should be set to 0640 with the
+        name of the owner being '{{ user }}' and the group being '{{ group }}'."
+  when: ( st_keystore.stat.mode != '0640' ) or ( st_keystore.stat.pw_name != '{{user}}' ) or ( st_keystore.stat.gr_name != '{{group}}' )
```

Comment:
This may be too conservative. Ideally we should look for **at least** read permission.